test/dynamodbmapper-v1/src/it/java/software/amazon/awssdk/services/dynamodb/mapper/TypeConvertedTimestampIntegrationTest.java

```diff
@@ -150,7 +156,7 @@ public void testInvalidPattern() throws Exception {
      */
     @DynamoDbTable(tableName = "aws-java-sdk-util")
     public static class KeyAndCalendarTimestamp extends AutoKeyAndVal<Calendar> {
-        @DynamoDbTypeConvertedTimestamp(pattern = "yyyyMMddHHmmssSSSz")
+        @DynamoDbTypeConvertedTimestamp(pattern = "yyyy MMddHHmmssSSSz")
```

Comment:
This is a bug in the JDK: https://bugs.openjdk.java.net/browse/JDK-8031085

Reply:
Does this do anything to how data is stored in Dynamo?

Reply:
Yes, the mapper would store the the formatted date using this format; so with the current mapper, users would not be able to use this the old pattern to store their dates.

Reply:
Hmm, kind of a deal breaker then. Wouldn't want unsuspecting customers to encounter this.

Any chance we can use the workaround described here:

https://bugs.openjdk.java.net/browse/JDK-8031085?focusedCommentId=13450079&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13450079

Reply:
Yeah it would be surprising behavior, but as the the mapper is no longer accessible in v2, there's currently no danger. 

The problem with the workaround is that we'd need to be able to parse the user-supplied format for 'SSS' and replace it with `appendValue(ChronoField.MILLI_OF_SECOND, 3)` which I don't want to do.

It seems like in the next iteration of the mapper, we should just just allow users to set the `DateTimeFormatter` to use instead, which would allow it to be much more flexible.